sklearn/__init__.py

```diff
@@ -17,6 +17,11 @@
 import warnings
 import os
 from contextlib import contextmanager as _contextmanager
+import logging
+
+logger = logging.getLogger(__name__)
```

Comment:
I guess this is the just the general convention right?

I found this from the [Python doc](https://docs.python.org/3.6/howto/logging.html#logging-basic-tutorial)
> A good convention to use when naming loggers is to use a module-level logger, in each module which uses logging, named as follows:

> logger = logging.getLogger(__name__)

and this from the [Hitchhiker's guide to Python](http://python-guide-pt-br.readthedocs.io/en/latest/writing/logging/#logging-in-a-library):

> Best practice when instantiating loggers in a library is to only create them using the __name__ global variable: the logging module creates a hierarchy of loggers using dot notation, so using __name__ ensures no name collisions.